### ISSUE TITLE
Can not see AWS instance details after upgrade to 1.20.6

### ISSUE THREAD

**Issue by diasjorge** [spinnaker/spinnaker#5929]
### Issue Summary:
When clicking on an AWS instance on the infrastructure tab, I see: Instance not found
GCE instances work correctly

### Cloud Provider(s):
AWS

### Environment:
I noticed this error after jumping from 1.19.12 to 1.20.6. I had to rollback and the problem went away

### Feature Area:
UI

### Description:
I see the response from the server is correct but on the browser console I see:
```
undefined is not an object (evaluating 'g.name')
```

### Steps to Reproduce:
Click on an instance deployed on AWS

**Comment by akshayabd** [comment]
Is this still an issue? Also, I'm guessing you did try force refreshing/deleting browser cache?

**Comment by diasjorge** [comment]
I tried updating Today to 1.21.4 and I saw the issue. I cleared my browser cache and even tried on Safari and Chrome.

Is there any more information I can give you to help diagnose the problem?

**Comment by diasjorge** [comment]
So I've debugged a bit more and found this error trace:
```
angular.js:14961 TypeError: Cannot read property 'name' of undefined
    at utils.ts:20
    at Array.find (<anonymous>)
    at utils.ts:20
    at Array.forEach (<anonymous>)
    at utils.ts:19
    at Array.forEach (<anonymous>)
    at applyHealthCheckInfoToTargetGroups (utils.ts:17)
    at instance.details.controller.js:68
    at instance.details.controller.js:192
    at angular.js:17330 "Possibly unhandled rejection: {}"
```

It seems somehow it thinks some of my targetGroups are "undefined"

```
(7) [undefined, undefined, {…}, {…}, {…}, {…}, {…}]
0: undefined
1: undefined
2: {account: "main-aws", cloudProvider: "aws", labels: {…}, moniker: {…}, name: "aws-lb1", …}
3: {account: "main-aws", cloudProvider: "aws", labels: {…}, moniker: {…}, name: "aws-lb2", …}
```

**Comment by diasjorge** [comment]
From this code: https://github.com/spinnaker/deck/blob/43c8075e07564f747fe75f0d72dd440e8c766112/app/scripts/modules/amazon/src/instance/details/utils.ts#L8

it seems to be trying to get the target groups from some load balancers from GCE.

```
0: {account: "account1-gce", certificate: "cert-2021", cloudProvider: "gce", createdTime: 1490030680619, defaultService: {…}, …}
1: {account: "account2-gce", certificate: "cert-2021", cloudProvider: "gce", createdTime: 1489431819653, defaultService: {…}, …}
2: {account: "main-aws", cloudProvider: "aws", labels: {…}, moniker: {…}, name: "aws-lb1", …}
3: {account: "main-aws", cloudProvider: "aws", labels: {…}, moniker: {…}, name: "aws-lb2", …}
```

**Comment by diasjorge** [comment]
So it seems the problem comes from this commit: https://github.com/spinnaker/deck/commit/ed7c44589cfbf482047a0f47d30df97165053aa4

My application is deployed to GCE and AWS so I think when here: https://github.com/spinnaker/deck/commit/ed7c44589cfbf482047a0f47d30df97165053aa4#diff-8e6e9acc7a85cea2f589f83e3b131f3aR76 they pass `app.loadBalancers.data` it also includes de load balancers from other cloud providers.

I think two fixes could be done, first is passing here only the load balancers with cloudProvider of "aws", the other is doing the filtering in the getAllTargetGroups function call. What do you think?

**Comment by diasjorge** [comment]
@akshayabd I see there was a fix for a similar issue, which I commented on, but I got the error on another part of the app. https://github.com/spinnaker/deck/blob/43c8075e07564f747fe75f0d72dd440e8c766112/app/scripts/modules/ecs/src/instance/details/instance.details.controller.js#L61 I think it makes sense to move the filtering to the getAllTargetGroups function

**Comment by diasjorge** [comment]
https://github.com/spinnaker/deck/pull/8416

**Comment by rampatnaik** [comment]
@spinnaker-release : can you merge or cherrypick the fix to 1.20.8 ?

**Comment by rampatnaik** [comment]
@spinnakerbot can you merge or cherrypick the fix to 1.20.8 ?
